src/ir_passes/await.ml

```diff
@@ -295,8 +338,8 @@ and c_exp' context exp k =
     letcont f (fun f ->
     letcont k (fun k ->
     match eff exp1 with
-    | T.Triv ->
-      varE k -*- (t_exp context exp1)
+    | T.Triv (* ->  FIXME
+      varE k -*- (t_exp context exp1) *)
```

Comment:
revert if possible.

Reply:
Turns outs this optimization no longer applies since the handler will always be needed in case we do an async expression with effect Triv (since we don't yet track the potential synchronous failure of async expressions in the effect system)